From ltsi-dev-bounces@lists.linuxfoundation.org Mon May 28 02:46:58 2012
From: "Shimoda, Yoshihiro" <yoshihiro.shimoda.uh@renesas.com>
Date: Mon, 28 May 2012 18:46:55 +0900
Subject: [LTSI-dev] [PATCH 03/21] drivercore: Add helper macro for platform_driver boilerplate
To: ltsi-dev@lists.linuxfoundation.org
Message-ID: <4FC3498F.9030400@renesas.com>


>From 0ac87d22b5e2f88fcaf9888496c8e7b1100ee87c Mon Sep 17 00:00:00 2001
From: Grant Likely <grant.likely@secretlab.ca>
Date: Wed, 5 Oct 2011 11:29:49 -0600
Subject: drivercore: Add helper macro for platform_driver boilerplate

For simple modules that contain a single platform_driver without any
additional setup code then ends up being a block of duplicated
boilerplate.  This patch adds a new macro, module_platform_driver(),
which replaces the module_init()/module_exit() registrations with
template functions.

Signed-off-by: Grant Likely <grant.likely@secretlab.ca>
Acked-by: Greg Kroah-Hartman <gregkh@suse.de>
Reviewed-by: Magnus Damm <magnus.damm@gmail.com>
Reviewed-by: Mark Brown <broonie@opensource.wolfsonmicro.com>
Reviewed-by: Stephen Boyd <sboyd@codeaurora.org>
(cherry picked from commit 940ab88962bc1aff3273a8356d64577a6e386736)

Conflicts:

	drivers/spi/spi-altera.c
	drivers/spi/spi-ath79.c
	drivers/spi/spi-atmel.c
	drivers/spi/spi-bfin-sport.c
	drivers/spi/spi-coldfire-qspi.c
	drivers/spi/spi-davinci.c
	drivers/spi/spi-dw-mmio.c
	drivers/spi/spi-ep93xx.c
	drivers/spi/spi-fsl-espi.c
	drivers/spi/spi-gpio.c
	drivers/spi/spi-imx.c
	drivers/spi/spi-mpc512x-psc.c
	drivers/spi/spi-mpc52xx-psc.c
	drivers/spi/spi-mpc52xx.c
	drivers/spi/spi-nuc900.c
	drivers/spi/spi-oc-tiny.c
	drivers/spi/spi-ppc4xx.c
	drivers/spi/spi-s3c24xx.c
	drivers/spi/spi-sh-msiof.c
	drivers/spi/spi-sh-sci.c
	drivers/spi/spi-stmp.c
	drivers/spi/spi-tegra.c
	drivers/spi/spi-ti-ssp.c
	drivers/spi/spi-xilinx.c
	drivers/tty/serial/of_serial.c
	include/linux/platform_device.h

Signed-off-by: Yoshihiro Shimoda <yoshihiro.shimoda.uh@renesas.com>
---
 drivers/spi/spi_sh.c |   13 +------------
 1 files changed, 1 insertions(+), 12 deletions(-)

diff --git a/drivers/spi/spi_sh.c b/drivers/spi/spi_sh.c
index e0343d4..70c8af9 100644
--- a/drivers/spi/spi_sh.c
+++ b/drivers/spi/spi_sh.c
@@ -524,18 +524,7 @@ static struct platform_driver spi_sh_driver = {
 		.owner = THIS_MODULE,
 	},
 };
-
-static int __init spi_sh_init(void)
-{
-	return platform_driver_register(&spi_sh_driver);
-}
-module_init(spi_sh_init);
-
-static void __exit spi_sh_exit(void)
-{
-	platform_driver_unregister(&spi_sh_driver);
-}
-module_exit(spi_sh_exit);
+module_platform_driver(spi_sh_driver);

 MODULE_DESCRIPTION("SH SPI bus driver");
 MODULE_LICENSE("GPL");
-- 
1.7.1

_______________________________________________
LTSI-dev mailing list
LTSI-dev@lists.linuxfoundation.org
https://lists.linuxfoundation.org/mailman/listinfo/ltsi-dev

